components/post_view/channel_intro_message/messages/off_topic.tsx

```diff
@@ -29,32 +28,18 @@ const OffTopicIntroMessage = ({
     usersLimit,
     boardComponent,
 }: Props) => {
+    const channelIsArchived = channel.delete_at !== 0;
     const isPrivate = channel.type === Constants.PRIVATE_CHANNEL;
-    const boardCreateButton = (
-        <BoardsButton
-            channel={channel}
-            boardComponent={boardComponent}
-        />
-    );
-    const children = (
-        <SetHeaderButton
-            channel={channel}
-        />
-    );
     const totalUsers = stats.total_users_count;
 
-    let setHeaderButton = null;
-    if (children) {
-        setHeaderButton = (
-            <ChannelPermissionGate
-                teamId={channel.team_id}
-                channelId={channel.id}
-                permissions={[isPrivate ? Permissions.MANAGE_PRIVATE_CHANNEL_PROPERTIES : Permissions.MANAGE_PUBLIC_CHANNEL_PROPERTIES]}
-            >
-                {children}
-            </ChannelPermissionGate>
-        );
-    }
+    const renderButtons = !channelIsArchived;
+    const permissions = [isPrivate ? Permissions.MANAGE_PRIVATE_CHANNEL_PROPERTIES : Permissions.MANAGE_PUBLIC_CHANNEL_PROPERTIES];
+    const boardCreateButton = renderButtons ? <BoardsButton boardComponent={boardComponent}/> : null;
+    const setHeaderButton = renderButtons ? (
+        <SetHeaderButton
+            channel={channel}
+            permissions={permissions}
+        />) : null;
```

Comment:
```suggestion
        />
    ) : null;
```

Reply:
See latest update. SetHeaderButton no longer needs permissions. and this const is now a one-liner